android-app/app/src/main/java/de/dhbw/ka/se/fibo/models/Cashflow.java

```diff
@@ -85,9 +83,19 @@ public void setItems(List<Item> items) {
     public int compareTo(Cashflow other) {
         // It is a Public API that it is sorted DESC
         int result = other.getTimestamp().compareTo(getTimestamp());
-        if(0 == result){
-            result = -1;
+
+        if (0 == result) {
+            result = other.getOverallValue().compareTo(getOverallValue());
+        }
+
+        if (0 == result) {
+            result = other.getPlace().compareTo(getPlace());
         }
+
+        if (0 == result) {
+            result = other.getCategory().compareTo(getCategory());
+        }
+
```

Comment:
Thanks for your concerns. I would argue that this is needed. It is a logic error that the `Comparable` interface isn't implemented sound here, which may result in hard-to-debug bugs. After a second look, it might be best to compare based on the cashflow id as well.

That this hasn't resulted in any noticeable problems yet is out of pure luck. The comparison now is at least better than it was before (just returning `-1` for anything is a logic error, as there was no `1` in the opposite case).

Reply:
No, it wouldn't.

As I said, it's better than it was before and I would rather not be nit-picky here.

Let's leave it here, so we can eventually get this merging…

Reply:
I dont think that it is an improvement if it does not work with newly added cashflows. It might be a code improvement but definitely not an improvement for the user experience (and possible upcomming demos). I dont think that you should dix things that are not part of your ticket and at the same time might create other issues/edge cases. In addition most of your Implementation is not needed anymore, when we can use the cashflowID. I would really appreciate it when you could wait with this deterministic improvement until the cashflows are also sent to the backend :)